src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      const index = Animal.alive.indexOf(this);
+
+      if (index > -1) {
+        Animal.alive.splice(index, 1);
+      }
+    }
```

Comment:
[CODE PERFORMANCE] - The checkHealth method uses nested loops (indexOf + splice) which is not efficient. Consider using the Array filter method to remove dead animals from the Animal.alive array. This can be done directly in the bite method of the Carnivore class after reducing the health of the herbivore, or in a separate method if needed.